src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BedSearchService.kt

```diff
@@ -148,15 +148,28 @@ class BedSearchService(
           return@validated fieldValidationError
         }
 
+        var premisesCharacteristicIds = listOf<UUID>()
```

Comment:
could we maybe refactor the code to avoid the need for a mutable premiseCharacteristicIds? e.g.

```
val premiseCharacteristicIds = premisesCharacteristicsNames?.let {
    val premisesCharacteristics = characteristicService.getCharacteristicsByNames(premisesCharacteristicsNames)
    premisesCharacteristicIds = premisesCharacteristics.filter { it.isActive && it.matches(ServiceName.temporaryAccommodation.value, "premises") }.map { it.id }.toList()
} ?: emptyList()
```